content/intro-to-storybook/svelte/en/composite-component.md

```diff
@@ -0,0 +1,256 @@
+    ...tasks.filter(t => t.state === "TASK_PINNED"),
+    ...tasks.filter(t => t.state !== "TASK_PINNED")
+  ];
+</script>
+{#if loading}
+  <div class="list-items">
+    <LoadingRow />
+    <LoadingRow />
+    <LoadingRow />
+    <LoadingRow />
+    <LoadingRow />
+  </div>
+{/if}
+
+{#if tasks.length === 0 && !loading}
+  <div class="list-items">
+    <div class="wrapper-message">
+      <span class="icon-check" />
+      <div class="title-message">You have no tasks</div>
+      <div class="subtitle-message">Sit back and relax</div>
+    </div>
+  </div>
+{/if}
+
+{#each tasksInOrder as task}
+  <Task {task} on:onPinTask on:onArchiveTask />
+{/each} 
+```
+
+The added markup results in the following UI:
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="/intro-to-storybook/finished-tasklist-states.mp4"
+    type="video/mp4"
+  />
+</video>
+
+Note the position of the pinned item in the list. We want the pinned item to render at the top of the list to make it a priority for our users.
+
+## Automated testing
+
+In the previous chapter we learned how to snapshot test stories using Storyshots. With `Task` there wasn’t a lot of complexity to test beyond that it renders OK. Since `TaskList` adds another layer of complexity we want to verify that certain inputs produce certain outputs in a way amenable to automatic testing. To do this we’ll create unit tests using [Jest](https://facebook.github.io/jest/) coupled with a test renderer such as [Enzyme](http://airbnb.io/enzyme/).
+
+![Jest logo](/intro-to-storybook/logo-jest.png)
+
+### Unit tests with Jest
+
+Storybook stories paired with manual visual tests and snapshot tests (see above) go a long way to avoiding UI bugs. If stories cover a wide variety of component use cases, and we use tools that ensure a human checks any change to the story, errors are much less likely.
+
+However, sometimes the devil is in the details. A test framework that is explicit about those details is needed. Which brings us to unit tests.
+
+In our case, we want our `TaskList` to render any pinned tasks **before** unpinned tasks that it has passed in the `tasks` prop. Although we have a story (`withPinnedTasks`) to test this exact scenario, it can be ambiguous to a human reviewer that if the component **stops** ordering the tasks like this, it is a bug. It certainly won’t scream **“Wrong!”** to the casual eye.
+
+So, to avoid this problem, we can use Jest to render the story to the DOM and run some DOM querying code to verify salient features of the output.
+
+Create a test file called `src/components/TaskList.test.js`. Here, we’ll build out our tests that make assertions about the output.
+
+```javascript
+// src/components/TaskList.test.js
+import TaskList from "./TaskList.svelte";
+import { render } from "@testing-library/svelte";
+import { withPinnedTasks } from "./TaskList.stories";
+describe("TaskList", () => {
+  it("renders pinned tasks at the start of the list", () => {
+    setTimeout(() => {
```

Comment:
I was getting into some errors, when the tests ran. More specifically getting undefined errors, so that's the reason of the timeout.  Going to move the code for a async/await pattern for better readability if you don't mind

Reply:
I think I managed to make it work on my branch, take a look at https://github.com/tmeasday/learnstorybook-svelte/commit/a197f3dd8cf73fecc541782cf668a0e49fda4b55